torch/_inductor/codegen/cpp.py

```diff
@@ -3903,6 +3904,128 @@ def can_fuse_vertical(self, node1, node2):
             self._can_fuse_horizontal_impl(node1, node2) and not node1.is_reduction()
         ) or self.can_fuse_vertical_outer_loop(node1, node2)
 
+    def optimize_loop_order(self, nodes):
+        """
+        Apply loop split optimization.
+        When one of the indexing_exprs contains a division, we eliminate the division by splitting the loop
+        to avoid non-contiguous loads, subject to the following conditions:
+            1. No reduction and no mudular index for all nodes.
+            2. Only one node's one indexing_exprs contains a division, according to this indexing_exprs,
+               we can get the dimension that needs to be split, and the split dimension is contiguous
+               in all other indexing_exprs.
+
+        For example, if the node's var_ranges: {z0: 2, z1: 9216, z2: 960} and indexing_exprs:
+        {'index0': 8847360*z0 + 960*z1 + z2, 'index1': 32*z0 + (z2//30), 'index2': z2},
+        we will spilt z2 -> 30*z2 + z3, then the node's var_ranges will be changed to
+        {z0: 2, z1: 9216, z2: 32, z3: 30} and indexing_exprs will be changed to
+        {'index0': 8847360*z0 + 960*z1 + 30*z2 + z3, 'index1': 32*z0 + z2, 'index2': 30*z2 + z3}.
+        """
+
+        def has_reduce_or_modular_index(node):
+            def has_modular_index(expr):
+                if isinstance(expr, ModularIndexing):
+                    return True
+                return any(has_modular_index(arg) for arg in expr.args)
+
+            return len(node.group[1][1]) != 0 or any(
+                has_modular_index(expr) for _, expr in node._body.indexing_exprs.items()
+            )
+
+        # No reduction and no mudular
+        if any(has_reduce_or_modular_index(node) for node in nodes):
+            return nodes
+
+        split_var = None
+        split_number = None
+        divide_index_name = None
+
+        def has_one_div(node):
```

Comment:
can we use visitor? See implementation in `simplify_index_in_vec_range`.

Reply:
I'm not sure I get your point, but I simplified this part of the code.

Reply:
The visitor way is not suitable here, because sympy.Wild can only be used in replace and match scenarios, not in the current find scenario.